libs/live-config/src/providers/jsonFileReader/index.ts

```diff
@@ -0,0 +1,33 @@
+import fs from "fs";
+import { Provider } from "..";
+import { ConfigInfo } from "../../LiveConfig";
+
+export class JsonFileReader implements Provider {
+  filePath: string;
+
+  constructor(config: { filePath: string }) {
+    if (!config.filePath.endsWith(".json")) {
+      throw new Error("Only .json files can be read by the provider");
+    }
+    this.filePath = config.filePath;
+  }
+
+  getValueBykey<K>(key: K, info: ConfigInfo) {
+    if (!fs.existsSync(this.filePath)) {
+      throw new Error("Invalid file path, could not read config");
+    }
+
+    try {
+      const fileContent = fs.readFileSync(this.filePath);
+      const parsedValue = JSON.parse(fileContent.toString());
+      return parsedValue[key] ?? info.default;
+    } catch (err) {
+      console.error(
+        `The config file doesn't have a key ${key}. Returning default value: ${info.default}`,
```

Comment:
We could also log this if JSON.parse fails

Maybe separate two cases : key doesn't exist, parsing error

Minor

Reply:
I think it doesn't matter a lot as the jsonFileReader is not used in production. It is just an example to show how to use this lib. We use only firebaseRemoteConfig as provider in production.

Reply:
Will only be used in unit tests